File: InnovateConnectSpring/backend/src/main/java/com/innovateconnect/api/models/Idea.java
package com.innovateconnect.api.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "ideas")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Idea {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @NotBlank(message = "Title is required")
    @Column(name = "title", nullable = false)
    private String title;

    @NotBlank(message = "Description is required")
    @Column(name = "description", nullable = false, columnDefinition = "TEXT")
    private String description;

    @NotBlank(message = "Technology used is required")
    @Column(name = "technology_used", nullable = false)
    private String technologyUsed;

    @Builder.Default
    @Column(name = "posted_date")
    private LocalDateTime postedDate = LocalDateTime.now();

    public String getStudentName() {
        return student != null ? student.getFullName() : "Anonymous";
    }
}

File: InnovateConnect/backend/InnovateConnect.API/Models/Idea.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace InnovateConnect.API.Models
{
    public class Idea
    {
        public int Id { get; set; }

        public int StudentId { get; set; }
        [ForeignKey("StudentId")]
        public Student? Student { get; set; }

        [Required]
        public string Title { get; set; } = string.Empty;

        [Required]
        public string Description { get; set; } = string.Empty;

        [Required]
        public string TechnologyUsed { get; set; } = string.Empty;

        public DateTime PostedDate { get; set; } = DateTime.UtcNow;
    }
}
